cli/src/typecheck.rs

```diff
@@ -9,11 +11,25 @@ use crate::{
 pub struct TypecheckCommand {
     #[command(flatten)]
     inputs: InputOptions<NoCustomizeMode>,
+
+    /// Start the type-checker in strict mode, so that the entire input is treated as a typed block.
+    #[arg(long, global = true)]
+    pub strict_typechecking: bool,
 }
 
 impl TypecheckCommand {
     pub fn run(self, global: GlobalOptions) -> CliResult<()> {
+        if self.strict_typechecking {
+            // In strict mode we run *both* forms of typechecking, because in fact neither one
+            // is more strict than the other.
```

Comment:
Nitpick: showing an example here (especially given that we have a very small one) could help a future reader here, otherwise this sentence could look a tad mystic.